pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -342,13 +342,21 @@ def on_save_checkpoint(
             "best_model_path": self.best_model_path,
             "current_score": self.current_score,
             "dirpath": self.dirpath,
+            "best_k_models": self.best_k_models,
+            "kth_best_model_path": self.kth_best_model_path,
+            "kth_value": self.kth_value,
+            "last_model_path": self.last_model_path,
         }
 
     def on_load_checkpoint(
         self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", callback_state: Dict[str, Any]
     ) -> None:
         self.best_model_score = callback_state["best_model_score"]
```

Comment:
Should this check that the same monitor is being used and only load in that case?

Reply:
@carmocca I think yes, but I also think that this might go beyond this PR and should be done in a separate PR. This PR is only a bugfix release and IMO your approach changes the behavior and would thereby go to feature releases.

Reply:
Oh right, I forgot about this magic :P